apps/condo/domains/marketplace/schema/RegisterInvoiceService.js

```diff
@@ -0,0 +1,150 @@
+const { Ticket } = require('@condo/domains/ticket/utils/serverSchema')
+
+const ERRORS = {
+    NO_INVOICE_CONTEXT: {
+        code: BAD_USER_INPUT,
+        type: ERROR_NO_INVOICE_CONTEXT,
+        message: 'The organization hasn\'t set up the marketplace',
+        messageForUser: 'api.marketplace.registerInvoice.error.noInvoiceContext',
+    },
+    ITEM_FROM_OTHER_ORGANIZATION: (rowNumber) => ({
+        code: BAD_USER_INPUT,
+        type: ERROR_ITEM_FROM_OTHER_ORGANIZATION,
+        message: `Item from other organization. Check line ${rowNumber}`,
+        messageForUser: 'api.marketplace.registerInvoice.error.itemFromOtherOrganization',
+        messageInterpolation: { rowNumber },
+    }),
+}
+
+const MOBILE_APP_RESIDENT_TICKET_SOURCE_ID = '830d1d89-2d17-4c5b-96d1-21b5cd01a6d3'
+
+const RegisterInvoiceService = new GQLCustomSchema('RegisterInvoiceService', {
+    schemaDoc: 'Using by mobile application. Allows residents to create invoice in pair with related ticket.',
+    types: [
+        {
+            access: true,
+            type: 'input InvoiceRowsInput { priceScope: MarketPriceScopeWhereUniqueInput!, count: Int! }',
+        },
+        {
+            access: true,
+            type: 'input RegisterInvoiceInput { dv: Int!, sender: SenderFieldInput!, resident: ResidentWhereUniqueInput!, invoiceRows: [InvoiceRowsInput!]! }',
+        },
+        {
+            access: true,
+            type: 'type RegisterInvoiceOutput { invoice: Invoice! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canRegisterInvoice,
+            schema: 'registerInvoice(data: RegisterInvoiceInput!): RegisterInvoiceOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { data } = args
+                const { dv, sender } = data
+                const userId = get(context, ['authedItem', 'id'])
+
+                const locale = extractReqLocale(context.req) || conf.DEFAULT_LOCALE
+
+                const resident = await getByCondition('Resident', { deletedAt: null, id: data.resident.id })
+                const invoiceContext = await getByCondition('InvoiceContext', {
+                    deletedAt: null,
+                    organization: { id: resident.organization },
+                    status: INVOICE_CONTEXT_STATUS_FINISHED,
+                })
+
+                if (!invoiceContext) {
+                    throw new GQLError(ERRORS.NO_INVOICE_CONTEXT, context)
+                }
+
+                const priceScopesCounts = {}
+                const priceScopesIds = data.invoiceRows.map((row) => {
+                    const id = get(row, ['priceScope', 'id'])
+                    set(priceScopesCounts, id, get(row, 'count', 0))
+
+                    return id
+                })
+
+                const priceScopes = await MarketPriceScope.getAll(context, { deletedAt: null, id_in: priceScopesIds })
+
```

Comment:
I think we don't. In this case, the `priceScopes` variable will not include a non-existent price scope.

Reply:
I can create an Invoice with empty `rows` if I specify `invoiceRows` with non-existent `priceScopeId`. Is it ok?

Would it be obvious that non-existent data is simply ignored rather than an error being returned?

Reply:
> I can create an Invoice with empty rows if I specify invoiceRows with non-existent priceScopeId. Is it ok?

ok, I saw that with empty `rows` you can only create a draft)

Reply:
> Would it be obvious that non-existent data is simply ignored rather than an error being returned?

How about adding a couple of lines about this to docs? :)

Reply:
I added the new throw if no rows are found before creating the ticket.

Reply:
Also, to ensure we do not create a ticket without invoice, we create invoice first.
Then we create a new ticket and connect the invoice.